core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -654,4 +650,12 @@ private void waitForAllFutures(List<Entry<String, Future<?>>> futureEntries)
             log.debug("{} successfully shut down.", entry.getKey());
         }
     }
+
+    private void waitForAllThreads() {
```

Comment:
Using `TestUtils.waitForCondition` is ok but we should move it out of `final` block. The infinite wait is not good in testing.

Reply:
Sorry for misunderstanding your last comment. I change back to `waitForCondition` and move `waitForAllThreads` out of `finally` block. Thank you.